deepchecks/base/check.py

```diff
@@ -395,23 +397,34 @@ def run(self, model) -> CheckResult:
         pass
 
 
-class ModelComparisonBaseCheck(BaseCheck):
-    """Parent class for check that compares between two or more models."""
-
-    def run(self,
-            train_datasets: Union[Dataset, List[Dataset]],
-            test_datasets: Union[Dataset, List[Dataset]],
-            models: List[Any]
-            ) -> CheckResult:
-        """Preprocess the parameters and pass them to `_run`."""
+class ModelComparisonContext:
+    """Contain processed input for model comparison checks."""
+    def __init__(self,
+                 train_datasets: Union[Dataset, List[Dataset]],
+                 test_datasets: Union[Dataset, List[Dataset]],
+                 models: Union[List[Any], Mapping[str, Any]]
+                 ):
```

Comment:
Why is this a class and not a function? If this is mainly validations...

Reply:
Also, why not use a context for every train-test-model triplet and then just send as a list? Only because of the names thing?

Reply:
because I want extra info, and some of this extra info is shared. for example task_type is single, not every triplet has its own. 
Also, the name makes it a quadruplet, and for every little thing you will want to add in the future it will change to quintuple, and then to sextuple, septuple, octuple, up to milluple.

Reply:
:+1: